### PR TITLE
feat: use dynamic credentials

### PR DESCRIPTION
**Pull request by mateuszjenek**
This PR replaces static cloud account to dynamic one: [GCP Service account impersonation](https://developer.humanitec.com/platform-orchestrator/security/cloud-accounts/gcp/#gcp-service-account-impersonation)

### CODE REVIEW

File: modules/base/variables.tf [hunk 1]
```diff
@@ -13,6 +13,12 @@ variable "region" {
   description = "GCP Region to provision resources in."
 }
 
+variable "humanitec_org_id" {
+  description = "Humanitec Organization ID (required for Backstage)"
```
Comment (johanneswuerbach) [inline modules/base/variables.tf:17]: This documentation is outdated now, right? The Organization ID is now required in any case.

(Same for for the root `variables.tf`)

File: modules/dynamic_creds/main.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
+data "google_project" "project" {
```
Comment (johanneswuerbach) [inline modules/dynamic_creds/main.tf:1]: 🤔  I wonder whether we should move this into https://github.com/humanitec-architecture/resource-packs-gcp as a standalone definition people could install and then re-use this here as well.

Comment (mateuszjenek) [reply modules/dynamic_creds/main.tf:1]: Hmm... I got your point, however this module does not include any resource definitions. Additionally, resource pack resources operate differently. They are provisioned via Humanitec only if there are workloads that depend on them. In contrast, this module creates a cloud account independently of Humanitec

File: modules/base/variables.tf (hunk 1 above)
Comment (mateuszjenek) [reply modules/base/variables.tf]: Fixed both!

File: modules/dynamic_creds/main.tf (hunk 2 above)
Comment (johanneswuerbach) [reply modules/dynamic_creds/main.tf:1]: 👍 , I'm mainly concerned that we need similar logic now in https://github.com/humanitec-architecture/resource-packs-gcp/blob/main/examples/dns/main.tf#L1-L29 (and other places where accounts might be used) and that customers don't have an easy way create a linked service-account.

In any case, could also be extracted as a follow-up.

**Review by johanneswuerbach — APPROVED** [review]

**Pull request merged**